### PR TITLE
Enable installing Galley with sidecar

### PR DESCRIPTION
**Pull request by myidpt**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ x ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml [hunk 1]
```diff
@@ -85,25 +94,93 @@ spec:
...
+        - name: istio-proxy
+      {{- if contains "/" .Values.global.proxy.image }}
+          image: "{{ .Values.global.proxy.image }}"
+      {{- else }}
+          image: "{{ .Values.global.hub }}/{{ .Values.global.proxy.image }}:{{ .Values.global.tag }}"
+      {{- end }}
+          imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+          ports:
+            - containerPort: 9902
+          {{- if not .Values.sidecar }}
```
Comment (morvencao) [inline install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml]: Will it be executed? given the while block is included in `{{- if .Values.sidecar }}`.

File: install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml [hunk 2]
```diff
@@ -42,9 +42,18 @@ spec:
...
+{{- if .Values.sidecar }}
+          - containerPort: 15019
+            protocol: TCP
+          - containerPort: 9443
+{{- else }}
           - containerPort: 443
+{{- end }}
+            protocol: TCP
           - containerPort: {{ .Values.global.monitoringPort }}
+            protocol: TCP
```
Comment (ayj) [inline install/kubernetes/helm/istio/charts/galley/templates/deployment.yaml]: `protocol` defaults to `TCP`. These can be omitted.

File: install/kubernetes/helm/istio/charts/galley/values.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ replicaCount: 1
 rollingMaxSurge: 100%
 rollingMaxUnavailable: 25%
 image: galley
+sidecar: true
```
Comment (ayj) [inline install/kubernetes/helm/istio/charts/galley/values.yaml]: Should we default to `false` until this has some soak time?

File: install/kubernetes/helm/istio/charts/galley/templates/service.yaml [hunk 4]
```diff
@@ -17,5 +17,7 @@ spec:
     name: http-monitoring
   - port: 9901
     name: grpc-mcp
+  - port: 15019
+    name: grpc-tls-mcp
```
Comment (ayj) [inline install/kubernetes/helm/istio/charts/galley/templates/service.yaml]: Does this mean the sidecar is only used for incoming mTLS connections? Don't we also want the envoy stats for clear text?

**Pull request closed without merging**